src/platform/backends/libvirt/libvirt_virtual_machine.cpp

```diff
@@ -269,12 +284,33 @@ void mp::LibVirtVirtualMachine::stop()
 
 void mp::LibVirtVirtualMachine::shutdown()
 {
-    virDomainShutdown(domain.get());
-    state = State::off;
-    update_state();
+    if (state == State::running || state == State::delayed_shutdown)
+    {
+        virDomainShutdown(domain.get());
+        state = State::off;
+        update_state();
+    }
+
     monitor->on_shutdown();
 }
 
+void mp::LibVirtVirtualMachine::suspend()
+{
+    if (state == State::running || state == State::delayed_shutdown)
+    {
+        if (virDomainManagedSave(domain.get(), 0) < 0)
+            throw std::runtime_error(virGetLastErrorMessage());
+
+        if (update_suspend_status)
+        {
+            state = State::suspended;
+            update_state();
+        }
+    }
+
+    monitor->on_suspend();
+}
+
```

Comment:
So we're NOOPing when it's shut down, but user asked to suspend? I wonder if that shouldn't be an error instead?

Reply:
Yep, missed that...but it should be handled in the daemon, so will fix that there.

Reply:
I fixed this in the recent push by adding some log messages when this situation occurs.